### PR TITLE
replace golang purge script with powershell script

### PR DESCRIPTION
**Pull request by ryanfkeepers**
## Description

Swap from GraphAPI-based data cleanup (which doesn't allow
us to access hidden folders like Audit retention or delete-restoration)
to powershell script-based cleanup (which allows us to make
SOAP requests against legacy exchange apis).

## Type of change

- [x] :bug: Bugfix
- [x] :robot: Test

## Issue(s)

* #1266

## Test Plan

- [x] :muscle: Manual
- [x] :green_heart: E2E

### CODE REVIEW

File: .github/actions/purge-m365-user-data/action.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+# * All folders in PersonMetadata 
+# * All already soft-deleted items
+# * All recoverable items in Audits
+# * All recoverable items in Purges
+
+inputs:
+  user:
+    description: User whose data is to be purged.
+  folder-prefix:
+    description: Name of the folder to be purged.  If falsy, will purge the set of static, well known folders instead.
```
Comment (gmatev) [inline .github/actions/purge-m365-user-data/action.yml:21]:
```suggestion
    description: Name of the folder to be purged.  If false, will purge the set of static, well known folders instead.
```

File: .github/workflows/ci_test_cleanup.yml [hunk 2]
```diff
@@ -8,17 +8,17 @@ jobs:
   Test-User-Data-Cleanup:
     environment: Testing
     runs-on: ubuntu-latest
+    continue-on-error: true
+    strategy:
+      matrix:
+        folder: [Corso_Restore_, TestRestore, '']
```
Comment (gmatev) [inline .github/workflows/ci_test_cleanup.yml:14]: Does this cover all calendars and contact patterns as well? I remember seeing some other event patterns last week along the lines to TestEvent, but maybe wrong.

**Review by gmatev — APPROVED** [review]
Looks good assuming all prefixes are covered

File: .github/workflows/ci_test_cleanup.yml (hunk 2 above)
Comment (gmatev) [reply .github/workflows/ci_test_cleanup.yml:14]: Here is some other patterns I see 

```
TestRestoreEvent:
TestRestoreEventObject:
```

Comment (ryanfkeepers) [reply .github/workflows/ci_test_cleanup.yml:14]: Should be covered by the prefix

Comment (gmatev) [reply .github/workflows/ci_test_cleanup.yml:14]: I don't see how. I think these were folders resulting from a restore test. There is no indication that they were wrapped in a Corso_Restore folder first

Output from Get-MailboxStatistics on the CI user
```
/Deleted Items                                                              19 403.8 KB (413,444 bytes)
/Deleted Items/Corso_Restore_28-Oct-2022_20:04:18                            2 76.42 KB (78,251 bytes)
/Deleted Items/Corso_Restore_28-Oct-2022_20-01-36.…                          2 76.45 KB (78,283 bytes)
/Deleted Items/Corso_Restore_28-Oct-2022_20-02-02.…                          3 40.71 KB (41,691 bytes)
/Deleted Items/Corso_Restore_28-Oct-2022_20-02-05.…                          5 67.85 KB (69,475 bytes)
/Deleted Items/TestRestoreEvent: 28-Oct-2022_20:22…                          1 39.75 KB (40,702 bytes)
/Deleted Items/TestRestoreEvent: 28-Oct-2022_20:36…                          1 39.77 KB (40,721 bytes)
/Deleted Items/TestRestoreEvent: 28-Oct-2022_20:43…                          1 10.46 KB (10,706 bytes)
/Deleted Items/TestRestoreEventObject: 28-Oct-2022…                          1 13.67 KB (13,993 bytes)
/Deleted Items/TestRestoreEventObject: 28-Oct-2022…                          1 13.67 KB (14,002 bytes)
/Deleted Items/TestRestoreEventObject: 28-Oct-2022…                          1 7.201 KB (7,374 bytes)
/Deleted Items/TestRestoreEventObject_28-Oct-2022_…                          1 17.82 KB (18,246 bytes)
```

**Pull request merged**